runtime/onert/core/src/exec/Executors.cc

```diff
@@ -21,68 +21,76 @@ namespace onert
+  return _executors.at(subg_index).get();
 }
 
-uint32_t Executors::inputSize() const
+uint32_t SingleModelExecutors::inputSize() const
 {
-  return _model_edges ? _model_edges->pkg_inputs.size()
-                      : entryExecutor()->graph().getInputs().size();
+  return entryExecutor()->graph().getInputs().size();
 }
 
-uint32_t Executors::outputSize() const
+uint32_t SingleModelExecutors::outputSize() const
 {
-  return _model_edges ? _model_edges->pkg_outputs.size()
-                      : entryExecutor()->graph().getOutputs().size();
+  return entryExecutor()->graph().getOutputs().size();
 }
 
-const ir::OperandInfo Executors::inputInfo(const ir::IOIndex &index)
+const ir::OperandInfo SingleModelExecutors::inputInfo(const ir::IOIndex &index)
 {
-  if (_model_edges)
-  {
-    auto const desc = _model_edges->pkg_inputs[index.value()];
-    auto const model_index = std::get<0>(desc);
-    auto const subg_index = std::get<1>(desc);
-    auto const executor = at(model_index, subg_index);
-    auto const input_index = executor->graph().getInputs().at(std::get<2>(desc));
-    return executor->graph().operands().at(input_index).info();
-  }
-
   const auto input_index = entryExecutor()->graph().getInputs().at(index);
   return entryExecutor()->graph().operands().at(input_index).info();
 }
 
-const ir::OperandInfo Executors::outputInfo(const ir::IOIndex &index)
+const ir::OperandInfo SingleModelExecutors::outputInfo(const ir::IOIndex &index)
 {
-  if (_model_edges)
-  {
-    auto const desc = _model_edges->pkg_outputs[index.value()];
-    auto const model_index = std::get<0>(desc);
-    auto const subg_index = std::get<1>(desc);
-    auto const executor = at(model_index, subg_index);
-    auto const output_index = executor->graph().getOutputs().at(std::get<2>(desc));
-    return executor->graph().operands().at(output_index).info();
-  }
-
   auto output_index = entryExecutor()->graph().getOutputs().at(index);
   return entryExecutor()->graph().operands().at(output_index).info();
 }
 
-void Executors::execute(const IODescription &desc)
+void SingleModelExecutors::execute(const IODescription &desc) { entryExecutor()->execute(desc); }
+
+void Executors::emplace(const ir::ModelIndex &model_index, const ir::SubgraphIndex &subg_index,
+                        std::unique_ptr<IExecutor> exec)
 {
-  if (_model_edges)
-    return executeModels(desc);
+  _executors.emplace(std::make_pair(model_index, subg_index), std::move(exec));
+}
 
-  entryExecutor()->execute(desc);
+IExecutor *Executors::at(const ir::ModelIndex &model_index,
+                         const ir::SubgraphIndex &subg_index) const
+{
+  return _executors.at(std::make_pair(model_index, subg_index)).get();
+}
+
+uint32_t Executors::inputSize() const { return _model_edges->pkg_inputs.size(); }
+
+uint32_t Executors::outputSize() const { return _model_edges->pkg_outputs.size(); }
+
+const ir::OperandInfo Executors::inputInfo(const ir::IOIndex &index)
```

Comment:
How about returning `const ir::OperandInfo &` if the purpose of the function is always a getter?

Reply:
Good. Let's change this later.